<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip" />
  <arg name="load_gripper" default="true" />
  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

 <param name="publish_frequency" value="50.0" type="double" />
  <node name="netft_node" pkg="netft_utils" type="netft_node" respawn="false" output="screen" args="192.168.1.1"/>

  <arg name="axis"  default="random"/>
  <param name="applied_axis" type="string" value="$(arg axis)"/>

  <arg name="desired_force"  default="0"/>
  <param name="FT_desired" type="double" value="$(arg desired_force)"/>

  <rosparam command="load" file="$(find franka_more_controllers)/config/franka_more_controllers.yaml"  />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="force_controller_invdynamics"/>
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_more_controllers)/launch/rviz/robot.rviz"/>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
  <node name="dynamic_reconfigure_compliance_param_node" pkg="dynamic_reconfigure" type="dynparam"
       args="set dynamic_reconfigure_compliance_param_node desired_force_torque $(arg desired_force)" />
</launch>
